testkit/src/main/scala/org/bitcoins/testkit/core/gen/ScriptGenerators.scala

```diff
@@ -49,6 +49,39 @@ sealed abstract class ScriptGenerators extends BitcoinSLogger {
     signatures.map(sigs => MultiSignatureScriptSignature(sigs))
   }
 
+  def conditionalScriptSignature: Gen[ConditionalScriptSignature] = {
+    val scriptSigGen = Gen
+      .oneOf(
+        packageToSequenceOfPrivateKeys(signedP2PKHScriptSignature),
+        packageToSequenceOfPrivateKeys(signedP2PKScriptSignature),
+        signedMultiSignatureScriptSignature
+        /* Can't have these since that would create nested Conditional(Timelock(Conditional))
+        signedCLTVScriptSignature,
+        signedCSVScriptSignature
+       */
+      )
+      .map(_._1)
+
+    scriptSigGen.flatMap { scriptSig =>
+      NumberGenerator.bool.map { condition =>
+        ConditionalScriptSignature(scriptSig, condition)
+      }
+    }
+  }
+
+  def nonLockTimeConditionalScriptSignature: Gen[ConditionalScriptSignature] = {
```

Comment:
scaladoc why it is useful not to include locktime

Reply:
I'm not sure what you mean?

This is only used by `randomNonLockTimeScriptSignature` which in turn is only used for the `CLTV` and `CSV` script signature generators in this file. (this is likely because TxBuilder does not currently handle nested locktimes and such correctly? idk why tbh)